addon/components/horizontal-bar-chart.js

```diff
@@ -230,13 +230,15 @@ export default ChartComponent.extend(FloatingTooltipMixin,
     };
   }),
 
-  valueLabelAttrs: Ember.computed('xScale', 'barThickness', 'labelPadding', function() {
+  valueLabelAttrs: Ember.computed('xScale', 'barThickness', 'labelPadding', 'hasNegativeValues', 'hasPositiveValues', function() {
     var xScale = this.get('xScale');
     // Anchor the label 'labelPadding' away from the zero line
     // How to anchor the text depends on the direction of the bar
+
     return {
       x: (d) => {
-        if (d.value < 0) {
+        //Value is negative, or Value is 0 and all other data is negative
+        if ((d.value < 0) || ((d.value === 0) && this.get('hasNegativeValues') && !this.get('hasPositiveValues'))) {
```

Comment:
Guys, stop working, it's thanksgiving.
On Nov 26, 2015 5:05 PM, "Phil Nachum" <notifications@github.com> wrote:

> In addon/components/horizontal-bar-chart.js
> <https://github.com/Addepar/ember-charts/pull/136#discussion_r46011933>:
>
> >      return {
> >        x: (d) => {
> > -        if (d.value < 0) {
> > +        //Value is negative, or Value is 0 and all other data is negative
> > +        if ((d.value < 0) || ((d.value === 0) && this.get('hasNegativeValues') && !this.get('hasPositiveValues'))) {
>
> You're welcome to change it now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Addepar/ember-charts/pull/136/files#r46011933>.
>